cpp/src/arrow/buffer.h

```diff
@@ -196,10 +248,31 @@ class ARROW_EXPORT Buffer {
   /// \brief Return the buffer's capacity (number of allocated bytes)
   int64_t capacity() const { return capacity_; }
 
+  /// \brief Whether the buffer is directly CPU-accessible
+  ///
+  /// If this function returns true, you can read directly from the buffer's
+  /// `data()` pointer.  Otherwise, you'll have to `View()` or `Copy()` it.
+  bool is_cpu() const { return is_cpu_; }
+
+  const std::shared_ptr<Device>& device() const { return memory_manager_->device(); }
+
+  const std::shared_ptr<MemoryManager>& memory_manager() const { return memory_manager_; }
+
   std::shared_ptr<Buffer> parent() const { return parent_; }
 
+  // Convenience functions
```

Comment:
>  (Because we can't define instance methods for std::shared_ptr<Buffer>?)

Yes.

> I think that instance methods are more convenient than static functions because user's code is shorter with instance methods:

Agreed, though `using ::arrow::Buffer` makes the code a bit less long.